### PR TITLE
WIP: Breaking/scoped elements

### PR DESCRIPTION
**Pull request by jorenbroekema**
TODO:
- Documentation / rationale on why/when to use scoped elements.

### CODE REVIEW

File: packages/validate/src/ValidateMixin.js [hunk 1]
```diff
@@ -24,7 +26,15 @@ function arrayDiff(array1 = [], array2 = []) {
 export const ValidateMixin = dedupeMixin(
   superclass =>
     // eslint-disable-next-line no-unused-vars, no-shadow
-    class ValidateMixin extends SyncUpdatableMixin(SlotMixin(superclass)) {
+    class ValidateMixin extends ScopedElementsMixin(SyncUpdatableMixin(SlotMixin(superclass))) {
+      static get scopedElements() {
+        return {
+          ...super.scopedElements,
+          // TODO: Remove once we can load it asynchronously, see todo below in loadFeedbackComponent method
```
Comment (daKmoR) [inline packages/validate/src/ValidateMixin.js]: we no longer need to load it asynchronously as we can override it when extending

Comment (jorenbroekema) [reply packages/validate/src/ValidateMixin.js]: True! Good catch :D

File: packages/input-datepicker/src/LionInputDatepicker.js [hunk 2]
```diff
@@ -350,7 +359,7 @@ export class LionInputDatepicker extends OverlayMixin(LionInputDate) {
     if (this._cachedOverlayContentNode) {
       return this._cachedOverlayContentNode;
     }
-    this._cachedOverlayContentNode = this.shadowRoot.querySelector('lion-calendar-overlay-frame');
+    this._cachedOverlayContentNode = this.shadowRoot.querySelector('.calendar__overlay-frame');
```
Comment (tlouisse) [inline packages/input-datepicker/src/LionInputDatepicker.js]: I would argue to not use class selectors (styles) for functional purposes (element references): https://css-tricks.com/stop-using-css-selectors-non-css/

Especially in our Lion layer, they can be swapped out by Sub classers.
For a practical example: see 'An extension implementation' here: https://github.com/ing-bank/lion/issues/591

Comment (jorenbroekema) [reply packages/input-datepicker/src/LionInputDatepicker.js]: I didn't see an alternative, I am aware classes are as a best practice for styling only, but as with all best practices, you only follow them if you can. If you can't you need to go with the lesser of multiple evils, which is either id or class, or do you propose something else?

Soon, scoped-elements will release a helper to do query selection on scoped elements by allowing to get the tagname from the scoped registry so we may use that, but I had to patch this somehow in the meantime

Comment (tlouisse) [reply packages/input-datepicker/src/LionInputDatepicker.js]: Let's go for the tag name from scoped-elements indeed. In other contexts, I would use a data attr (or alternatively a class with 'js' prefix) like the article suggests

**Comment by CubLion** [comment]
Still to be done:
- select-rich
- switch

File: packages/input-datepicker/src/LionInputDatepicker.js [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 
 /**
  * @customElement lion-input-datepicker
  * @extends {LionInputDate}
  */
-export class LionInputDatepicker extends OverlayMixin(LionInputDate) {
+export class LionInputDatepicker extends ScopedElementsMixin(OverlayMixin(LionInputDate)) {
+  static get scopedElements() {
+    return {
+      ...super.scopedElements,
```
Comment (CubLion) [inline packages/input-datepicker/src/LionInputDatepicker.js]: `scopedElements` is static. Does this super call make sense?

Comment (jorenbroekema) [reply packages/input-datepicker/src/LionInputDatepicker.js]: Yes, super refers to the parent constructor which can access static methods and properties.

**Comment by daKmoR** [comment]
implemented via https://github.com/ing-bank/lion/pull/629

**Pull request closed without merging**